fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,637 @@
+        # Also check that events and confound files can be loaded as DataFrame
+        if events is not None:
+            events = _check_run_tables(run_imgs, events, "events")
+        if confounds is not None:
+            confounds = _check_run_tables(run_imgs, confounds, "confounds")
+
+        # For each run fit the model and keep only the regression results.
+        nistats_flm.labels_, nistats_flm.results_, self.design_matrices_ = ([], [], [])
+        n_runs = len(run_imgs)
+        t0 = time.time()
+        for run_idx, run_img in enumerate(run_imgs):
+            # Report progress
+            if nistats_flm.verbose > 0:
+                percent = float(run_idx) / n_runs
+                percent = round(percent * 100, 2)
+                dt = time.time() - t0
+                # We use a max to avoid a division by zero
+                if run_idx == 0:
+                    remaining = "go take a coffee, a big one"
+                else:
+                    remaining = (100.0 - percent) / max(0.01, percent) * dt
+                    remaining = "%i seconds remaining" % remaining
+
+                sys.stderr.write(
+                    "Computing run %d out of %d runs (%s)\n"
+                    % (run_idx + 1, n_runs, remaining)
+                )
+
+            # Build the experimental design for the glm
+            run_img = check_niimg(run_img, ensure_ndim=4)
+            if design_matrices is None:
+                n_scans = run_img.get_data().shape[3]
+                if confounds is not None:
+                    confounds_matrix = confounds[run_idx].values
+                    if confounds_matrix.shape[0] != n_scans:
+                        raise ValueError(
+                            "Rows in confounds does not match"
+                            "n_scans in run_img at index %d" % (run_idx,)
+                        )
+                    confounds_names = confounds[run_idx].columns.tolist()
+                else:
+                    confounds_matrix = None
+                    confounds_names = None
+                start_time = nistats_flm.slice_time_ref * nistats_flm.t_r
+                end_time = (n_scans - 1 + nistats_flm.slice_time_ref) * nistats_flm.t_r
+                frame_times = np.linspace(start_time, end_time, n_scans)
+                design = make_first_level_design_matrix(
+                    frame_times,
+                    events[run_idx],
+                    nistats_flm.hrf_model,
+                    nistats_flm.drift_model,
+                    nistats_flm.high_pass,
+                    nistats_flm.drift_order,
+                    nistats_flm.fir_delays,
+                    confounds_matrix,
+                    confounds_names,
+                    nistats_flm.min_onset,
+                )
+            else:
+                design = design_matrices[run_idx]
+            self.design_matrices_.append(design)
+
+            img_mat = run_img.get_data()
```

Comment:
`get_data()` is being deprecated. If you want data scaled according to the scale factors:

```suggestion
            img_mat = run_img.get_fdata()
```

If you want the raw data block, as found on-disk:


```suggestion
            img_mat = run_img.dataobj.get_unscaled()
```

If you want the old `get_data()` behavior where what you get is dependent on whether the scale factors are non-identity:


```suggestion
            img_mat = np.asanyarray(run_img.dataobj)
```

Reply:
i think I just want the scaled data.